src/index.js

```diff
@@ -1,39 +1,114 @@
 import './style.css';
+import DoTask from './task.js';
+import './delete.png';
+import './action.png';
+import './refresh.png';
 
 const listContainer = document.querySelector('.list-container');
+const inputDo = document.getElementById('todo-input');
 
-const finalData = [
-  {
-    description: 'hello',
-    completed: false,
-    index: 0,
-  },
-  {
-    description: 'time is up',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'how is it',
-    completed: false,
-    index: 2,
-  },
-];
+// class
+
+const task = new DoTask();
 const updateData = () => {
   let html = '';
-  for (let i = 0; i < finalData.length; i += 1) {
-    if (finalData[i].completed === false) {
+  for (let i = 0; i < task.tasks.length; i += 1) {
+    if (task.tasks[i].completed === false) {
       html += `<div class="item">
-        <div class="item-detail">
-            <input type="checkbox" class="item-check">
-            <h5>${finalData[i].description}</h5>
-        </div>
-        <img src="./action.png" id="${i}" alt="" class="dot">
-    </div>`;
+      <div class="main-item">
+          <div class="item-detail">
+              <input type="checkbox" class="item-check" id="${task.tasks[i].index}">
+              <h5 class="descr">${task.tasks[i].description}</h5>
+          </div>
+          <img src="./action.png"  alt="" class="dot" disabled>
+      </div>
+      <div class="main-edit">
+          <input type="checkbox" class="checkbox-edit" id="">
+          <input type="text" class="item-input-edit" value="${task.tasks[i].description}" id="${task.tasks[i].index}">
+          <img src="./delete.png" alt="" class="delete-item" id="${task.tasks[i].index}">
+      </div>
+  </div>`;
     }
   }
   listContainer.innerHTML = html;
   listContainer.classList.add('text-gray');
+  const allData = document.querySelectorAll('.item-check');
+  const descriptionAll = document.querySelectorAll('.descr');
+  for (let j = 0; j < allData.length; j += 1) {
+    allData[j].addEventListener('change', function () {
+      if (allData[j].checked) {
+        descriptionAll[j].classList.add('strike-through');
+        const btnId = parseInt(this.id, 10);
+        const myArray = task.tasks;
+        const myObject = myArray.find((obj) => obj.index === btnId);
+        myObject.completed = true;
+        const updateArray = JSON.stringify(myArray);
+        task.updateComplete(updateArray);
+      } else {
+        descriptionAll[j].classList.remove('strike-through');
+        const btnId = parseInt(this.id, 10);
+        const myArray = task.tasks;
+        const myObject = myArray.find((obj) => obj.index === btnId);
+        myObject.completed = false;
+        const updateArray = JSON.stringify(myArray);
+        task.updateComplete(updateArray);
+      }
```

Comment:
- Please ensure the property index is set to the value of the new array length (i.e., if we have 4 tasks on the list, the index of the tasks should be 1 - 4 and not 0 -3), Please fix this.
![Screenshot from 2023-03-04 14-00-20](https://user-images.githubusercontent.com/44624138/222903352-e4cc3175-b1fd-4ebb-8aa5-ebc83f129f51.png)